### PR TITLE
Added more power operations for domain mode server, added tests for them.

### PR DESCRIPTION
**Pull request by hhovsepy**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 6fb002d5eaea2010895890d6fabe6af2aa9518d8:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8caeab180469b80d6dc55a278e82d097247646ae:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 23946c6c3b6afbb2e7a2dea23bfbd829c099bb4b:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1042a98d896e575e1dd8c36ab5392152286671c1:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/middleware/server_methods.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+from utils.wait import wait_for
+from cfme.middleware.server import MiddlewareServer
+
+DELAY = 20
+NUM_SEC = 700
+
+
+def check_server_stopped(provider, server):
```
Comment (psav) [inline cfme/tests/middleware/server_methods.py]: Perhaps these should be renamed ```wait_for.....``` as they do more than just the check.

File: cfme/tests/middleware/test_middleware_server.py [hunk 2]
```diff
@@ -19,11 +21,19 @@
...
     yield server
     server.restart_server()
 
 
+@pytest.yield_fixture(scope="function")
+def domain_server(provider):
+    server = get_server_by_name(provider, EAP_PRODUCT_NAME, 'EAP7-server-one')
+    yield server
+    server.start_server()
+    server.restart_server()
```
Comment (psav) [inline cfme/tests/middleware/test_middleware_server.py]: What's the need for the restart here?

Comment (psav) [inline cfme/tests/middleware/test_middleware_server.py]: Probably deserves a comment

Comment (hhovsepy) [inline cfme/tests/middleware/test_middleware_server.py]: the reason to restart was that 'start' does not help if server is started but suspended, better solution is to resume and then start, it covers all cases and is not so 'hard' as restart.

File: cfme/tests/middleware/server_methods.py (hunk 1 above)
Comment (hhovsepy) [inline cfme/tests/middleware/server_methods.py]: I think better is 'verify_server_...', wdyt ?

**Comment by dajoRH** [comment]
Lint report for commit f97abdf5e6d76e51291185d4909932faf47b0210:

Everything seems all right from lint perspective :smile: .

Commit inspection report:

**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by hhovsepy** [comment]
failed 57z are because of some sever EAP7 side problems

**Pull request merged**